gene2phenotype_project/gene2phenotype_app/views/user.py

```diff
@@ -98,49 +104,155 @@ class CreateUserView(generics.CreateAPIView):
     permission_classes = [permissions.IsAuthenticated]
 
 
-class LoginView(KnoxLoginView):
+class LoginView(generics.GenericAPIView):
+    """
+        LoginView: Handles user authentication and login.
+
+        This view allows users to authenticate using their credentials (e.g., email/username and password).
+        Upon successful authentication, the view generates a JSON Web Token (JWT) for the user and 
+        sets it as a secure cookie in the HTTP response. 
+
+        Attributes:
+            - serializer_class: Specifies the serializer used for validating user credentials, 
+            which is `LoginSerializer`.
+            - permission_classes: Allows any user (authenticated or not) to access this endpoint 
+            by using the `AllowAny` permission.
+
+        Methods:
+            - post(request, *args, **kwargs): Handles the user login process, generates the 
+            access token, and sets it in a cookie.
+    """
+
+    serializer_class = LoginSerializer
+    permission_classes = (permissions.AllowAny,)
+    def post(self, request, *args, **kwargs):
+        """
+            Handles the POST request for user login.
+
+            This method authenticates a user based on the provided credentials, generates an access 
+            token, and sets it in a secure cookie for client-side use. It also returns the user 
+            details and token information in the response body.
+            Parameters:
+                - request: The HTTP request containing user credentials in the request body (e.g., 
+                `username` and `password`).
+            Returns:
+                - Response: An HTTP response object with the user data and tokens in the body. 
+                The access token is also stored as a secure cookie.
+                The refresh token is also stored as a secure cookie
+
+            Raises:
+                - ValidationError: Raised if the provided credentials are invalid.
+                - ValueError: Raised if the cookie cannot be set due to a response error.
+        """
+
+        serializer = LoginSerializer(data=request.data)
```

Comment:
The context is missing:
`serializer = LoginSerializer(data=request.data, context={'request': request})`
This is being fetched in the LoginSerializer